### ISSUE TITLE
Incorporate Staleness parameter

### ISSUE THREAD

**Issue by UditGupta10** [uber/horovod#631]
As the 'optimizer' is wrapped around 'DistributedOptimizer', gradient averaging occurs after each batch computation. I am using horovod for word2vec kind of problem where we have sparse gradient updates and it has been brought up in other issues that the overall time for multinode increases due to excessive communication overhead.
For this reason, I thought of synchronizing after 'n' batch computations rather than the default 1.
Does horovod support it?
If not could you point me in the right direction for its implementation.
(My understanding is that if DistributedOptimizer is executed after 'n' batch computations then the synchronisation will happen after n batch computations).

**Comment by tgaddair** [comment]
Hey @UditGupta10, not yet, but there's an active pull request #546 that adds this functionality for PyTorch.

**Comment by UditGupta10** [comment]
My request is specific to Tensorflow.

**Comment by alsrgv** [comment]
@andfoy and we are still figuring the right way to do it for PyTorch, mathematically speaking.  Once all that is nailed down, we can make it work in TensorFlow, too.

**Comment by UditGupta10** [comment]
Appreciate the work you guys are doing. Hope you figure it out soon. Please let me know when you are through.
Thanks

**Comment by alsrgv** [comment]
@UditGupta10, please feel free to subscribe to #546 for updates.
